relay-server/src/services/processor/span/processing.rs

```diff
@@ -135,14 +144,18 @@ pub fn process(
             item.set_metrics_extracted(true);
         }
 
-        // TODO: dynamic sampling
+        if let Some(sampling_outcome) = &sampling_outcome {
+            relay_log::trace!(
+                "Dropping span because of sampling rule {}",
+                sampling_outcome
+            );
+            return ItemAction::Drop(sampling_outcome.clone());
+        }
```

Comment:
Untested:

```suggestion
        if let Some(sampling_outcome) = sampling_outcome {
            relay_log::trace!(
                "Dropping span because of sampling rule {}",
                sampling_outcome
            );
            return ItemAction::Drop(sampling_outcome);
        }
```

You leave the branch here, which means you should be able to take ownership of the value.

Reply:
This returns from `retain_items`, but the sampling outcome is defined outside the loop.